fundingmanager_test.go

```diff
@@ -1226,7 +1232,19 @@ func TestFundingManagerFundingNotTimeoutInitiator(t *testing.T) {
 			len(pendingChannels))
 	}
 
-	recreateAliceFundingManager(t, alice)
+	// Restart the funding manager in the background, which ensures that
+	// rebroadcasting the funding txn will not block the test.
+	go recreateAliceFundingManager(t, alice)
```

Comment:
If you make `publishChan` buffered, then this wouldn't be necessary (also the following sleep could be removed),

Reply:
Noice, fixed!

Reply:
Actually, I still think the sleep is still necessary. There's a race condition when trying to read on `publTxChan`

```
==================
WARNING: DATA RACE
Read at 0x00c4202abbe0 by goroutine 22:
  github.com/lightningnetwork/lnd.TestFundingManagerFundingNotTimeoutInitiator()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/fundingmanager_test.go:1244 +0x298
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:746 +0x16c
Previous write at 0x00c4202abbe0 by goroutine 112:
  github.com/lightningnetwork/lnd.recreateAliceFundingManager()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/fundingmanager_test.go:392 +0xc05
Goroutine 22 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:921 +0x206
  main.main()
      github.com/lightningnetwork/lnd/_test/_testmain.go:96 +0x1d3
Goroutine 112 (running) created at:
  github.com/lightningnetwork/lnd.TestFundingManagerFundingNotTimeoutInitiator()
      /home/travis/gopath/src/github.com/lightningnetwork/lnd/fundingmanager_test.go:1237 +0x279
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.4.linux.amd64/src/testing/testing.go:746 +0x16c
==================
```

Reply:
Actually ignore this, accidentally took this from the og version 😅